### ISSUE TITLE
[Question] Using custom Roslyn dlls under custom built VS Next

### ISSUE THREAD

**Issue by Hsn723** [dotnet/roslyn#20538]
I successfully built the Roslyn project with some of my added syntax and am running VisualStudioSetup.Next in which I loaded a test project which uses the syntax I added. I wanted to include the Microsoft.CodeAnalysis.CSharp (and etc.) DLLs that were built in my project to write a SyntaxWalker to capture nodes using my new syntax. Upon building that test project, I get the following error:
```
System.IO.FileLoadException: 'Could not load file or assembly 'Microsoft.CodeAnalysis.CSharp, Version=42.42.42.42, Culture=neutral, PublicKeyToken=31bf3856ad364e35'. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)'

```
The DLLs were loaded from `C:\Users\antoine\AppData\Local\Microsoft\VisualStudio\15.0_90ef690dRoslynDev\Extensions\Microsoft\Roslyn Language Services\42.42.42.42424`

Am I doing something wrong or unsupported?

**Comment by Pilchie** [comment]
Can you provide more information about *exactly* how you mean by: "running VisualStudioSetup.Next", and how you referenced things in your test project, etc?

**Comment by NatsumiHoshino** [comment]
Hi, thanks for the reply. What I did was load the Roslyn project on VS, apply my changes and after making sure it builds properly I selected VisualStudioSetup.Next as the startup project and ran it with F5 in Debug (and Release) configurations.

Once on the experimental VS instance that gets launched from it, I created a test console project (.NET Core), tried to add the `Microsoft.CodeAnalysis.dll`, `Microsoft.CodeAnalysis.CSharp.dll` etc. that were built above using Add Reference on the test project and browsing to the DLLs. Then, I wrote a CSharpSyntaxWalker for the new syntax that I added. Then, when attempting to launch the test project in the experimental VS instance with F5 I run into the error message I mentioned previously. The DLLs are referenced in my test project's .csproj file like so:
```
<ItemGroup>
    <Reference Include="Microsoft.CodeAnalysis">
      <HintPath>..\..\..\Users\antoine\AppData\Local\Microsoft\VisualStudio\15.0_90ef690dRoslynDev\Extensions\Microsoft\Roslyn Language Services\42.42.42.42424\Microsoft.CodeAnalysis.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.CodeAnalysis.CSharp">
      <HintPath>..\..\..\Users\antoine\AppData\Local\Microsoft\VisualStudio\15.0_90ef690dRoslynDev\Extensions\Microsoft\Roslyn Language Services\42.42.42.42424\Microsoft.CodeAnalysis.CSharp.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.CodeAnalysis.CSharp.Workspaces">
      <HintPath>..\..\..\Users\antoine\AppData\Local\Microsoft\VisualStudio\15.0_90ef690dRoslynDev\Extensions\Microsoft\Roslyn Language Services\42.42.42.42424\Microsoft.CodeAnalysis.CSharp.Workspaces.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.CodeAnalysis.Workspaces">
      <HintPath>..\..\..\Users\antoine\AppData\Local\Microsoft\VisualStudio\15.0_90ef690dRoslynDev\Extensions\Microsoft\Roslyn Language Services\42.42.42.42424\Microsoft.CodeAnalysis.Workspaces.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.CodeAnalysis.Workspaces.Desktop">
      <HintPath>..\..\..\Users\antoine\AppData\Local\Microsoft\VisualStudio\15.0_90ef690dRoslynDev\Extensions\Microsoft\Roslyn Language Services\42.42.42.42424\Microsoft.CodeAnalysis.Workspaces.Desktop.dll</HintPath>
    </Reference>
  </ItemGroup>
```

**Comment by Pilchie** [comment]
Interesting.  I'm not sure where else it's finding those assemblies from, but my guess is that it's from the .NET Core Shared Runtime.

Can you:
1. Opt out of using the shared runtime?
2. Make sure that those references are copied into your output directory (may need to set CopyLocal on them in the properties window)?

@srivatsn - any ideas here for the .NET Core aspect of this?

**Comment by Pilchie** [comment]
BTW - you can just reference the binaries out of your `Binaries\Debug\Dlls\` directory after building Roslyn - no need to go rummaging around in %LocalAppData%.

**Comment by Hsn723** [comment]
Hi,
I have made sure that the references are copied to the output directory. I am not sure how to opt out of using the shared runtime though. I didn't see that option in the `dotnet` command line options or in VS. Though looking under Dependencies -> SDK of my Solution, I see a Microsoft.CodeAnalysis (1.3.0) under `Microsoft.NETCore.App`. Could that be the problem?

Also, I am not sure if it helps, but I tried recreating the test console project as a classic console app targeting .NET Framework 4.6 instead of using .NET Core. After manually importing some of the missing DLLs that `Microsoft.CodeAnalysis.dll` depended on I was able to run my test project successfully.

I am curious why I can't get it to run under .NET Core, but if that's not possible at the moment using .NET Framework is a good enough interim workaround.
